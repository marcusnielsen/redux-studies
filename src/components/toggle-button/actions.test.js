import actionTypes from '../action-types'
import Actions from './actions'

test('actions initial state', function onTest () {
  const actions = Actions({
    dispatch: () => {},
    actionsTypes: 'a',
    entity: 'b',
    id: 'c',
    field: 'd'
  })

  expect(Object.keys(actions)).toMatchSnapshot()
})

test('actions.update', function onTest (done) {
  function dispatch (data) {
    expect(data).toMatchSnapshot()
    done()
  }

  const actions = Actions({
    dispatch,
    actionTypes,
    entity: 'b',
    id: 'c',
    field: 'd'
  })

  actions.update(true)
})
