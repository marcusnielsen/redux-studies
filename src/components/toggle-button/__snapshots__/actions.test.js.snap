exports[`test actions initial state 1`] = `
Array [
  "update",
]
`;

exports[`test actions.update 1`] = `
Object {
  "entity": "b",
  "field": "d",
  "id": "c",
  "payload": Object {
    "active": true,
  },
  "type": "update",
}
`;
